File: leetcode/src/java/BestTimeToBuyAndSellStock.java
public class BestTimeToBuyAndSellStock {
    public static void main(String[] args) {
        BestTimeToBuyAndSellStock instance = new BestTimeToBuyAndSellStock();
        System.out.println(instance.maxProfit(new int[]{7,1,5,3,6,4}));
        System.out.println(instance.maxProfit(new int[]{7,6,4,3,1}));
    }
    public int maxProfit(int[] prices) {
        int result = 0;
        int lowest = 10000;
        for (int price : prices) {
            if (lowest > price) {
                lowest = price;
            } else {
                if (price - lowest > result) {
                    result = price - lowest;
                }
            }
        }
        return result;
    }
}

File: leetcode/src/python/best_time_to_buy_and_sell_stock.py
from typing import List

class BestTimeToBuyAndSellStock:
    def maxProfit(self, prices: List[int]) -> int:
        min_price = -1
        result = 0
        for price in prices:
            if min_price == -1:
                min_price = price
            elif price < min_price:
                min_price = price
            elif price - min_price > result:
                result = price - min_price
        return result